### PR TITLE
Dirtybit

### PR DESCRIPTION
**Pull request by sippeangelo**
Dirty flags now exist for all component fields if they're ever requested by `ComponentWrapper::Dirty`. Dirty flag categories can be separated by the hardcoded enum `DirtySetType`

``` cpp
enum class DirtySetType
{
    Transform,
    Network
};
```

As a result of this change, you can no longer obtain mutable references to component fields. The old syntax of requesting a const reference to an enum field still works, along with assignment by value, but if you want to store a reference to a field and modify it later you need to use the proxy class `Field`.
### Example

``` cpp
ComponentWrapper cTransform = entity["Transform"];
// Const reference (quickest method)
const glm::vec3& scale = cTransform["Scale"];
// Value assignment
cTransform["Position"] = glm::vec3(1.f, 2.f, 3.f);
// Mutable field
Field<glm::vec3> position = cTransform["Position"];
position += glm::vec3(1.f);
position.y(position.y() + 3); // position.y += 3;
```

Basically, everywhere you would have written `double&` you now have to write `Field<double>` to be able to modify it later.

### CODE REVIEW

**Comment by sippeangelo** [comment]
[Everything Is AWESOME](https://www.youtube.com/watch?v=JwQZQygg3Lk)
